homeassistant/components/netatmo/__init__.py

```diff
@@ -36,6 +56,7 @@
         vol.Required(CONF_PASSWORD): cv.string,
         vol.Required(CONF_SECRET_KEY): cv.string,
         vol.Required(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_WEBHOOK_URL): cv.string,
```

Comment:
This is the wrong approach. Netatmo should just generate it's own webhook url and pass it to the integration. The user should not be involved with that at all.

You can generate IDs with:
 -  `hass.components.webhook.async_generate_id`
 -  `hass.components.webhook.async_generate_url`
 -  `hass.components.webhook.async_register`

Reply:
But that's now how it works with Netatmo. You have to hard-code an URL on their dev-site. Generating an id/url in HASS would require to update the settings for the "app" every time a new id gets generated. I assume their implementation is targeted at static hosts where they push the data to when the camera generates it. Even though the URL is included when subscribing, it has to be configured in the app as well.

Reply:
This would be a lot easier when you use a config entry, that way a webhook ID can be generated and stored and presented to the user when adding the integration.

Reply:
Or wait, maybe I'm wrong about this. I'll investigate further.

Reply:
You were right. It is possible to use dynamic webhooks. The documentation wasn't clear about this, but after some testing I'm fairly confident dynamic URLs work too.
I did change the config-option to be `webhooks: true`. They should be enabled manually because not everybody exposes hass. And I don't know if subscribing without being able to receive events is a good idea. According to the documentation at least the webhook-function gets banned after 5 times without a 200-response. I don't know how this could impact other parts of the API.